guacamole-common/src/main/java/org/apache/guacamole/protocol/ConfiguredGuacamoleSocket.java

```diff
@@ -59,17 +68,87 @@
      */
     private GuacamoleProtocolVersion protocolVersion =
             GuacamoleProtocolVersion.VERSION_1_0_0;
-    
+
+    /**
+     * Parses the given "error" instruction, throwing a GuacamoleException that
+     * corresponds to its status code and message.
+     *
+     * @param instruction
+     *     The "error" instruction to parse.
+     *
+     * @throws GuacamoleException
+     *     A GuacamoleException that corresponds to the status code and message
+     *     present within the given "error" instruction.
+     */
+    private static void handleReceivedError(GuacamoleInstruction instruction)
+            throws GuacamoleException {
+
+        GuacamoleStatus status;
+
+        // Consider all error instructions without a corresponding status code
+        // to be server errors
+        List<String> args = instruction.getArgs();
+        if (args.size() < 2) {
+            logger.debug("Received \"error\" instruction without status code.");
+            status = GuacamoleStatus.SERVER_ERROR;
+        }
+
+        else {
+
+            // Parse the status code from the received error instruction
+            int statusCode;
+            try {
+                statusCode = Integer.parseInt(args.get(1));
+            }
+            catch (NumberFormatException e) {
+                logger.debug("Received \"error\" instruction with non-numeric status code.", e);
+                return;
```

Comment:
Are you sure ignoring the instruction is the right thing to do here? This arguably represents a completely corrupt connection that should be aborted outright. Just my opinion, of course...

Reply:
This should probably be included in the release notes though (or added explicitly to the exception signature) to reflect the API refinement.

Reply:
Good catch! That's copy-pasta. That `return` definitely should not be there.

Reply:
Corrected via refactor and rebase.